### PR TITLE
audit: check bind for development versions

### PR DESCRIPTION
**Pull request by javian**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----
Add an audit that would have prevented https://github.com/Homebrew/homebrew-core/pull/40343 
Not sure if there's a guideline about where to place the audit in the file.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> Not sure if there's a guideline about where to place the audit in the file.

Somewhere around the existing ones at https://github.com/Homebrew/brew/blob/a5a18faae757945b441ac1216bef58fb8d9a5db0/Library/Homebrew/dev-cmd/audit.rb#L735 would be preferable. Thanks!

**Comment by MikeMcQuaid** [comment]
@javian Sorry, to be more specific: can you add a check that matches those ones e.g. a new `when` in `audit_specs`, thanks.

**Comment by javian** [comment]
I changed it up now and placed it where you asked.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -719,6 +719,22 @@ def audit_specs
...
+      when /bind/
+        version = Version.parse(stable.url)
+        return if version.to_s.split(".").second.to_i.even?
+
+        problem "BIND releases with odd minor version numbers (9.13.x, 9.15.x, etc) are " \
+        "for testing, and can be unstable and are not suitable for general deployment. " \
+      end
+    end
+
+    def audit_bind
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: Remove this.

**Comment by MikeMcQuaid** [comment]
Made a few tweaks. Thanks @javian!

**Pull request merged**